hazelcast/src/main/java/com/hazelcast/nio/Address.java

```diff
@@ -174,14 +178,12 @@ public boolean equals(Object o) {
             return false;
         }
         final Address address = (Address) o;
-        return port == address.port && this.type == address.type && this.host.equals(address.host);
+        return this.type == address.type && this.socketAddr.equals(address.socketAddr);
```

Comment:
Loos like `this.type == address.type` part is not really needed here:
```
     * Two instances of {@code InetAddress} represent the same IP
     * address if the length of the byte arrays returned by
     * {@code getAddress} is the same for both, and each of the
     * array components is the same for the byte arrays.
```
IPv4 address can't be equal to IPv6 address and we always (?) have non-null `InetAddress` provided to `InetSocketAddress`, if I understand the javadoc/code correctly.

Reply:
:+1: You are right. I'll fix it.